python/Ganga/GPIDev/Lib/File/File.py

```diff
@@ -146,10 +147,11 @@ class ShareDir(GangaObject):
 
     """
     _schema = Schema(Version(1, 0), {'name': SimpleItem(defvalue='', getter="_getName", doc='path to the file source'),
-                                     'subdir': SimpleItem(defvalue=os.curdir, doc='destination subdirectory (a relative path)')})
+                                     'subdir': SimpleItem(defvalue=os.curdir, doc='destination subdirectory (a relative path)'),
+                                     'associated_files': GangaFileItem(defvalue=[], typelist = [str, IGangaFile], doc='A list of files associated with the sharedir')})
 
     _category = 'shareddirs'
-    _exportmethods = ['add', 'ls', 'path']
+    _exportmethods = ['add', 'ls', 'path', 'removeAssociatedFiles', 'remove']
```

Comment:
Not really other than in case anyone finds it useful. Perhaps if you want to remove the uploaded files without deleting the job itself for record keeping.